src/components/UploadVideoComponent.tsx

```diff
@@ -1,44 +1,159 @@
+            allowsEditing: true,
+        });
+
+        if (video.cancelled === true) {
+            return;
+        }
+
+        setVideoUri(video.uri);
+        setVideoName("video selected");
+    };
+
+    /**
+     * Expo document picker that only allows picking of video files in mp4 format.
+     */
+    const documentPicker = async () => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let document: any = {};
+        document = await DocumentPicker.getDocumentAsync({
+            type: "video/mp4",
+        });
+
+        if (document.type === "cancel") {
+            return;
+        }
+
+        setVideoUri(document.uri);
+        setVideoName(document.name);
+    };
+
+    /**
+     * Upload the picked video to the backend.
+     * If no video was selected previously do
+     */
+    const uploadVideo = async () => {
+        if (videoUri === "") {
+            return;
+        }
+
+        const video = await buildVideoAsFormData();
+        const postRequest = RequestFactory.createPostRequestWithFormData(video);
+        const response = await endpointsVideo.uploadVideo(postRequest);
+
+        resetVideoState();
+        createAlert(i18n.t("itrex.uploadVideoSuccessMsg"));
+        return response;
+    };
+
+    /**
+     * Build a FormData object from the video uri.
+     */
+    const buildVideoAsFormData = async () => {
+        const fileBlob = await (await fetch(videoUri)).blob();
+        const formData = new FormData();
+        formData.append("file", fileBlob);
+        return formData;
+    };
+
+    const resetVideoState = () => {
+        setVideoUri("");
+        setVideoName("");
+    };
+
     return (
         <ImageBackground source={require("../constants/images/Background_forest.svg")} style={styles.image}>
             <Header title={i18n.t("itrex.toUploadVideo")} />
             <View style={styles.container}>
                 <View style={styles.StyledInputContainer}>
                     <Text>{i18n.t("itrex.uploadVideoHere")}</Text>
-                    <TextInput style={styles.StyledTextInput}></TextInput>
+                    <TextInput
+                        style={styles.StyledTextInput}
+                        value={videoName}
+                        editable={false}
+                        testID="videoNameInput"></TextInput>
+                    <Pressable style={styles.StyledButton}>
+                        <Button title={i18n.t("itrex.browseVideos")} onPress={pickVideo}></Button>
+                    </Pressable>
                 </View>
                 <Pressable style={styles.StyledButton}>
-                    <Button title={i18n.t("itrex.browseFiles")} onPress={changeStyle}></Button>
+                    <Button title={i18n.t("itrex.toUploadVideo")} onPress={uploadVideo}></Button>
                 </Pressable>
             </View>
         </ImageBackground>
     );
-
-    function changeStyle() {
-        return undefined;
-    }
```

Comment:
I believe this is fine since changeStyle() seems like an ancient placeholder function to me, correct me if I'm wrong.